### PR TITLE
[KYUUBI #2379][SUB-TASK][KPIP-4] Implement BatchesResource DELETE /batches/${batchId}

### LINKED ISSUE #2379: [SUB-TASK][KPIP-4] Implement BatchesResource DELETE /batches/${batchId}
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the feature

#2240

### Motivation

_No response_

### Describe the solution

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

To close #2379

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate
image

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by turboFei** [comment]
need verify the permission to kill job

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/BatchesResource.scala [hunk 1]
```diff
@@ -63,6 +65,43 @@ private[v1] class BatchesResource extends ApiRequestContext with Logging {
...
+    try {
+      val sessionHandle = sessionManager.getBatchSessionHandle(batchId, REST_BATCH_PROTOCOL)
+      session = sessionManager.getSession(sessionHandle).asInstanceOf[KyuubiBatchSessionImpl]
+    } catch {
+      case NonFatal(e) =>
+        error(s"Invalid batchId: $batchId", e)
+        throw new NotFoundException(s"Invalid batchId: $batchId")
+    }
+
+    // TODO: Support proxy user
```
Comment (ulysses-you) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/BatchesResource.scala:89]: is there a ticket for this todo ?

Comment (turboFei) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/BatchesResource.scala:89]: created #2556

Comment (turboFei) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/BatchesResource.scala:112]: How about?
```
      @PathParam("batchId") batchId: String,
      @QueryParam("killApp") killApp: Boolean,
      @QueryParam("proxyUser") proxyUser: String): Response = {
```

Comment (turboFei) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/BatchesResource.scala:112]: cc @ulysses-you

Comment (ulysses-you) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/BatchesResource.scala]: can it be `hive.server2.proxy.user` ?

**Review by ulysses-you — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks, merging to master

**Pull request closed without merging**